src/udev/udevadm-monitor.c

```diff
@@ -41,9 +41,9 @@ static void print_device(struct udev_device *device, const char *source, int pro
         struct timespec ts;
 
         assert_se(clock_gettime(CLOCK_MONOTONIC, &ts) == 0);
-        printf("%-6s[%"PRI_TIME".%06ld] %-8s %s (%s)\n",
+        printf("%-6s[%"PRI_TIME".%06lu] %-8s %s (%s)\n",
                source,
-               ts.tv_sec, ts.tv_nsec/1000,
+               ts.tv_sec, (unsigned long)ts.tv_nsec/1000,
```

Comment:
Yes, but we will also end up with the result not necessarily matching the format specifier which is the original problem this PR tries to address.